### PR TITLE
Migrate user groups

### PR DESCRIPTION
**Pull request by Pulya10c**
**🟢 Add `deploy` label if you want to deploy this Pull Request to staging environment**

#### 🧑‍⚖️ Pull Request Naming Convention

- Title should follow [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary)
- Do not put issue id in title
- Do not put WIP in title. Use [Draft PR](https://github.blog/2019-02-14-introducing-draft-pull-requests/) functionality
- Consider to add `area:*` label(s)

* [x] I followed naming convention rules

---

#### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Performance optimization
- [x] Refactoring
- [ ] Test Case
- [ ] Documentation update
- [ ] Other

#### 🔗 Related issue link

[_Describe the source of requirement, like related issue link._](https://github.com/rolling-scopes/rsschool-app/issues/1271)

#### 💡 Background and solution

_Describe the big picture of your changes here_

#### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [ ] Database migration is added or not needed
- [ ] Documentation is updated/provided or not needed
- [ ] Changes are tested locally

### CODE REVIEW

File: nestjs/src/userGroups/user-groups.controller.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import { ApiOkResponse, ApiOperation, ApiTags } from '@nestjs/swagger';
+import { DefaultGuard, RoleGuard } from 'src/auth';
+import { CreateUserGroupDto, UpdateUserGroupDto } from './dto';
+import { UserGroupDto } from './dto/user-group.dto';
+import { UserGroupsService } from './user-groups.service';
+
+@Controller('user-group')
+@ApiTags('user-group')
+@UseGuards(DefaultGuard, RoleGuard)
+export class UserGroupsController {
```
[the remark this answers is not in the document]
Comment (Pulya10c) [reply nestjs/src/userGroups/user-groups.controller.ts:11]: Yes

File: nestjs/src/userGroups/user-groups.service.ts [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        id: user.id,
+        githubId: user.githubId,
+        name: [user.firstName, user.lastName].filter(Boolean).join(' '),
+      };
+    };
+  }
+
+  private formatGroup(userGroups: UserGroup[], users: User[]) {
+    const usersGroups: UserGroupDto[] = userGroups
+      .map(g => ({
```
[the remark this answers is not in the document]
Comment (Pulya10c) [reply nestjs/src/userGroups/user-groups.service.ts]: Fixed

File: nestjs/src/userGroups/user-groups.controller.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  constructor(private readonly service: UserGroupsService) {}
+
+  @Post()
+  @ApiOperation({ operationId: 'createUserGroup' })
+  @ApiOkResponse({ type: UserGroupDto })
+  public async create(@Body() dto: CreateUserGroupDto) {
+    const [data] = await this.service.create(dto);
```
[the remark this answers is not in the document]
Comment (Pulya10c) [reply nestjs/src/userGroups/user-groups.controller.ts]: fixed

File: nestjs/src/userGroups/user-groups.controller.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (aaliakseyenka) [reply nestjs/src/userGroups/user-groups.controller.ts:11]: then u are missing  `@RequiredRoles([Role.Admin])`

**Review by aaliakseyenka — APPROVED** [review]

File: nestjs/src/userGroups/user-groups.controller.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (Pulya10c) [reply nestjs/src/userGroups/user-groups.controller.ts:12]: Thanks, fixed

**Review by apalchys — APPROVED** [review]

**Pull request merged**